evalml/tests/component_tests/test_per_column_imputer.py

```diff
@@ -212,7 +213,7 @@ def test_fit_transform_drop_all_nan_columns():
         pd.DataFrame(
             {
                 "all_nan": [np.nan, np.nan, np.nan],
-                "some_nan": [np.nan, 1, 0],
+                "some_nan": [0.0, 1.0, 0.0],
```

Comment:
Why did this need to change? Also, we should rename this column since there are no longer nans

Reply:
Sorry I thought I had answered this yesterday.

The only way to pass an all-NaN column to the imputer is if it's not typed as `Unknown`, which is what the default woodwork inference returns. So we need to init woodwork on the dataframe. However, because of https://github.com/alteryx/evalml/issues/2055, the imputer will modify the input data. 

So the `some_nan` column will have the NaNs imputed. We can't change the name of the column because the intention is to compare the input data before and after running through the imputer and the column name doesn't change after going through the imputer.